components/post_view/post_info/post_info.tsx

```diff
@@ -21,116 +23,125 @@ import PostReaction from 'components/post_view/post_reaction';
-
-        /*
-         * Funciton called when the post options dropdown is opened
-         */
-        handleDropdownOpened: PropTypes.func.isRequired,
-
-        /*
-         * Set to mark the post as flagged
-         */
-        isFlagged: PropTypes.bool,
-
-        /*
-         * Set to mark the post as open the extra info in the rhs
-         */
-        isCardOpen: PropTypes.bool,
-
-        /*
-         * The number of replies in the same thread as this post
-         */
-        replyCount: PropTypes.number,
-
-        /**
-         * Set to indicate that this is previous post was not a reply to the same thread
-         */
-        isFirstReply: PropTypes.bool,
-
-        /**
-         * Set to render in mobile view
-         */
-        isMobile: PropTypes.bool,
-
-        /**
-         * Set to render in compact view
-         */
-        compactDisplay: PropTypes.bool,
-
-        /**
-         * Set to mark post as being hovered over
-         */
-        hover: PropTypes.bool.isRequired,
-
-        /**
-         * Set to render the post time when not hovering
-         */
-        showTimeWithoutHover: PropTypes.bool.isRequired,
-
-        /**
-         * Whether to show the emoji picker.
-         */
-        enableEmojiPicker: PropTypes.bool, // Made it optional until migrating this file to TS
+export type Props = {
+
+    /**
+     * The post to render the info for
+     */
+    post: Post,
+
+    /**
+     * The id of the team which belongs the post
+     */
+    teamId?: string;
+
+    /**
+     * Function called when the comment icon is clicked
+     */
+    handleCommentClick: React.EventHandler<React.MouseEvent>;
+
+    /**
+     * Function called when the card icon is clicked
+     */
+    handleCardClick: (post: Post) => void;
+
+    /**
+     * Function called when the post options dropdown is opened
+     */
+    handleDropdownOpened: (e: boolean) => void;
+
+    /**
+     * Set to mark the post as flagged
+     */
+    isFlagged?: boolean;
+
+    /**
+     * Set to mark the post as open the extra info in the rhs
+     */
+    isCardOpen?: boolean;
+
+    /**
+     * The number of replies in the same thread as this post
+     */
+    replyCount?: number;
+
+    /**
+     * Set to indicate that this is previous post was not a reply to the same thread
+     */
+    isFirstReply?: boolean;
+
+    /**
+     * Set to render in mobile view
+     */
+    isMobile?: boolean;
+
+    /**
+     * Set to render in compact view
+     */
+    compactDisplay?: boolean;
+
+    /**
+     * Set to mark post as being hovered over
+     */
+    hover: boolean;
+
+    /**
+     * Set to render the post time when not hovering
+     */
+    showTimeWithoutHover: boolean;
+
+    /**
+     * Whether to show the emoji picker.
+     */
+    enableEmojiPicker?: boolean; // TechDebt: Made it optional due to post_header
+
+    /**
+     * Set not to allow edits on post
+     */
+    isReadOnly?: boolean | null;
```

Comment:
Can this be `null` as well as `undefined`?

Reply:
fixing in the next commit